data/shared/src/main/scala/sigma/data/ExactNumeric.scala

```diff
@@ -30,6 +32,66 @@ trait ExactNumeric[T] {
   def toInt(x: T): Int = n.toInt(x)
   def toLong(x: T): Long = n.toLong(x)
 
+  /** Returns a big-endian representation of this value in a collection of bytes.
+    * For example, the `Int` value `0x12131415` would yield the
+    * collection of bytes [0x12, 0x13, 0x14, 0x15]
+    */
+  def toBigEndianBytes(x: T): Coll[Byte]
+
+  /**
+    * Returns a big-endian binary representation of this value as boolean array.
+    */
+  def toBits(x: T): Coll[Boolean] = {
+
+    def isBitSet(byte: Byte)(bit: Int): Boolean = ((byte >> bit) & 1) == 1
+
+    def byte2Bools(b: Byte): Array[Boolean] =  (0 to 7).toArray.reverse.map(isBitSet(b))
+
+    val bytes = toBigEndianBytes(x)
+    val l = bytes.length
+    val res = new Array[Boolean](l * 8)
+    var offset = 0
+    cfor(0)(_ < l, _ + 1) { i =>
+      val b = bytes(i)
+      val bits = byte2Bools(b)
```

Comment:
It is possible to allocate `bits` once and then reuse as a buffer. 
Or even better, put boolean values directly into `res`.
It can be filled from inner cfor loop over bit indexes `cfor(7, _ >= 0, _ - 1)`.

Reply:
done